docs/configs-spec2006-tests/run_spec.py

```diff
@@ -0,0 +1,231 @@
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors: Jason Lowe-Power, Ayaz Akram, Hoa Nguyen
+
+""" Script to run a SPEC benchmark in full system mode with gem5.
+
+    Inputs:
+    * This script expects the following as arguments: name of the detailed
+    CPU, SPEC benchmark name, size of the benchmark, the path to the folder
+    to write SPEC log files, and paths to the kernel and the disk image.
+    * The SPEC benchmark name and the output folder must be provided, while 
+    the rest are optional.
+    * The default detailed CPU is KVM (not a detailed CPU).
+    * The default size is ref.
+    * The default path to the kernel is linux-stable/vmlinux-4.19.83
+    * The default path to the disk image is disk-image/spec2006/spec2006-image/spec2006
+
+    Outputs:
+    * TODO: simple performance statistics
+"""
+import os
+import sys
+import time
+
+import m5
+import m5.ticks
+from m5.objects import *
+
+import argparse
+
+from system import MySystem
+
+
+def writeBenchScript(dir, benchmark_name, size, output_path):
+    """
+    This method creates a script in dir which will be eventually
+    passed to the simulated system (to run a specific benchmark
+    at bootup).
+    """
+    input_file_name = '{}/run_{}_{}'.format(dir, benchmark_name, size)
+    with open(input_file_name,"w") as f:
+        f.write('{} {} {}'.format(benchmark_name, size, output_path))
+    return input_file_name
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="gem5 config file to run SPEC benchmarks")
+    parser.add_argument("-c", "--cpu", type = str,
+                        default = "kvm", help =" Name of the detailed CPU")
+    parser.add_argument("-b", "--benchmark", type = str, required = True,
+                        help = "Name of the SPEC benchmark")
+    parser.add_argument("-s", "--size", type = str, 
+                        default="ref", help = "Available sizes: test, train, ref")
+    parser.add_argument("-o", "--output-dir", type = str, required = True,
+                        help = "Folder to write SPEC logs")
+    parser.add_argument("-k", "--kernel", type = str,
+                        default = "linux-stable/vmlinux-4.19.83",
+                        help = "Path to vmlinux")
+    parser.add_argument("-d", "--disk", type = str,
+                        default = "disk-image/spec2006/spec2006-image/spec2006",
+                        help = "Path to the disk image containing SPEC benchmarks")
+    parser.add_argument("-l", "--copy-logs",
+                        action = "store_true", help = "Copy SPEC run logs to the host system")
+    parser.add_argument("-z", "--no-listeners",
+                        action = "store_true", help = "Turn off ports")
+    args = parser.parse_args()
+    
+    return args
+
+def getDetailedCPUModel(cpu_name):
+    available_models = {"kvm": X86KvmCPU,
+                        "o3": DerivO3CPU,
+                        "minor": MinorCPU}
```

Comment:
This might not compile if the MinorCPU isn't compiled. 

The CPUList is a better way to do this. However, I'm not sure this is simple for us to use. We should put this in a list of things we want to have in a gem5 python library.
https://gem5.googlesource.com/public/gem5/+/master/configs/common/ObjectList.py#157